lib/modules/versioning/ruby/index.spec.ts

```diff
@@ -293,23 +313,66 @@ describe('modules/versioning/ruby/index', () => {
     ${'v1.0.3'}              | ${'replace'}         | ${'1.0.3'}     | ${'1.2.3'}   | ${'v1.2.3'}
     ${'= 1.0.3'}             | ${'replace'}         | ${'1.0.3'}     | ${'1.2.3'}   | ${'= 1.2.3'}
     ${'!= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'!= 1.0.3'}
+    ${'!= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.0.2'}   | ${'!= 1.0.3'}
+    ${'!= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.0.3'}   | ${'!= 1.0.3'}
+    ${'> 1.0.3'}             | ${'replace'}         | ${'1.0.4'}     | ${'1.2.3'}   | ${'> 1.0.3'}
+    ${'> 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.0.3'}   | ${'> 1.0.2'}
     ${'< 1.0.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.2.4'}
+    ${'< 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.0.3'}   | ${'< 1.2.3'}
     ${'< 1.2.2'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.2.4'}
     ${'< 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.2.4'}
     ${'< 1.2'}               | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'< 1.3'}
     ${'< 1'}                 | ${'replace'}         | ${'0.9.0'}     | ${'1.2.3'}   | ${'< 2'}
     ${'< 1.2.3'}             | ${'replace'}         | ${'1.0.0'}     | ${'1.2.2'}   | ${'< 1.2.3'}
     ${'>= 1.0.3'}            | ${'replace'}         | ${'1.0.3'}     | ${'1.2.3'}   | ${'>= 1.0.3'}
+    ${'>= 1.0.3'}            | ${'replace'}         | ${'1.0.3'}     | ${'1.0.2'}   | ${'>= 1.0.2'}
     ${'<= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'<= 1.2.3'}
     ${'<= 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.0.2'}   | ${'<= 1.0.3'}
-    ${'~> 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'~> 1.2.0'}
+    ${'~> 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.2.3'}   | ${'~> 1.2.3'}
     ${'~> 1.0.3'}            | ${'replace'}         | ${'1.0.0'}     | ${'1.0.4'}   | ${'~> 1.0.3'}
     ${'~> 4.7, >= 4.7.4'}    | ${'replace'}         | ${'1.0.0'}     | ${'4.7.9'}   | ${'~> 4.7, >= 4.7.4'}
+    ${'~> 4.7, >= 4.7.4'}    | ${'replace'}         | ${'4.7.5'}     | ${'4.8.0'}   | ${'~> 4.7, >= 4.7.4'}
     ${'>= 2.0.0, <= 2.15'}   | ${'replace'}         | ${'2.15.0'}    | ${'2.20.1'}  | ${'>= 2.0.0, <= 2.20.1'}
-    ${'~> 5.2.0'}            | ${'replace'}         | ${'5.2.4.1'}   | ${'6.0.2.1'} | ${'~> 6.0.0'}
+    ${'~> 5.2.0'}            | ${'replace'}         | ${'5.2.4.1'}   | ${'6.0.2.1'} | ${'~> 6.0.2'}
```

Comment:
I would have expected `~> 6.0.0` here

Reply:
How about this one?

Reply:
This is also fixed in https://github.com/renovatebot/renovate/pull/17091/commits/037ac4237e400d5d09ae51e9e5940e8f731ecc8e.